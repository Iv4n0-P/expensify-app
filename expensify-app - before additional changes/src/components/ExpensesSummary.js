import React from 'react'
import selectExpensesTotal from '../selectors/expenses-total'
import SelectExpenses from '../selectors/expenses'
import { connect } from 'react-redux'
import numeral from 'numeral'
import { Link } from 'react-router-dom'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formatedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                    <Link className="button-blue" to="/create">Add Expense</Link>
                </div>
             </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expensesToShow = SelectExpenses(state.expenses, state.filters)
    return {
        expensesTotal: selectExpensesTotal(expensesToShow),
        expenseCount: expensesToShow.length
    }
}

export default connect(mapStateToProps)(ExpensesSummary)


