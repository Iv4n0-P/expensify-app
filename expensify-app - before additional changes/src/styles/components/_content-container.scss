.content-container {
    max-width: 80rem;
    margin: $l-size auto;
    padding: 0 $m-size;
}