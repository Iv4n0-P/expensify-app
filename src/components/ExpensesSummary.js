import React from 'react'
import selectExpensesTotal from '../selectors/expenses-total'
import SelectExpenses from '../selectors/expenses'
import { connect } from 'react-redux'
import numeral from 'numeral'
import { Link } from 'react-router-dom'

export const ExpensesSummary = ({ expenseCount, expensesTotal, hiddenExpenses, allExpenses }) => {
    const expenseWord = allExpenses === 1 ? 'expense' : 'expenses'
    const expenseWord2 = hiddenExpenses === 1 ? 'expense' : 'expenses'
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> of {allExpenses} {expenseWord} totalling <span>{formatedExpensesTotal}</span></h1>
                <p>{hiddenExpenses !==0 && `${hiddenExpenses} hidden ${expenseWord2}`}</p>
                <div className="page-header__actions">
                    <Link className="button-blue" to="/create">Add Expense</Link>
                </div>
             </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expensesToShow = SelectExpenses(state.expenses, state.filters)
    return {
        expensesTotal: selectExpensesTotal(expensesToShow),
        expenseCount: expensesToShow.length,
        allExpenses: state.expenses.length,
        hiddenExpenses: state.expenses.length - expensesToShow.length
    }
}

export default connect(mapStateToProps)(ExpensesSummary)


