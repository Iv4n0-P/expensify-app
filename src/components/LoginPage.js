import React from 'react'
import { connect } from 'react-redux'
import { startLogin, startLoginFacebook, startLoginTwitter } from '../actions/auth'

export const LoginPage = ({ startLogin, startLoginFacebook }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <button id='startLogin' className="button-blue button-google" onClick={startLogin}>Login with Google</button>
            <button id='startLoginFacebook' className="button-blue button-facebook" onClick={startLoginFacebook}>Login with Facebook</button>
            <button id='startLoginTwitter' className="button-blue button-twitter" onClick={startLoginTwitter}>Login with Twitter</button>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
    startLoginFacebook: () => dispatch(startLoginFacebook()),
    startLoginTwitter: () => dispatch(startLoginTwitter())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)