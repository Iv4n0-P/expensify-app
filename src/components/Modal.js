import React from 'react'
import Modal from 'react-modal'

export const OptionModal = (props) => (

  <Modal
    isOpen={props.showModal}
    contentLabel="Confirm deletion"
    ariaHideApp={false}
    className='modal'
  >
    <div>
      <p>Do you want to delete this expense?</p>
      <button className="button-blue button-github" onClick={props.onRemove}>Remove Expense</button>
      <button className="button-blue button-github" onClick={props.handleCloseModal}>Close</button>
    </div>
  </Modal>
)