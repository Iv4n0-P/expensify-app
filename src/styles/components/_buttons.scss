.button-blue {
    padding: $s-size $xl-size;
    background: $blue;
    color: white;
    display: inline-block;
    border-width: 0px;
    //ili border: none;
    font-size: $font-size-large; //čisto font u varijablu se može isto
    text-decoration: none;
    line-height: 1;
}

.button-blue--nobg {
    background: none;
    border: none;
    color: #f7f7f7;
}

.button-blue--secondary {
    background: #888888;
}

.button-google {
    width: 100%;
    background: $google;
    font-size: $font-size-medium;
    margin-top: 5px;
}

.button-facebook {
    margin-top: 5px;
    width: 100%;
    background: $facebook;
    font-size: $font-size-medium;
}

.button-twitter {
    margin-top: 5px;
    width: 100%;
    background: $twitter;
    font-size: $font-size-medium;
}

.button-github {
    margin-top: 5px;
    width: 100%;
    background: $github;
    font-size: $font-size-medium;
}