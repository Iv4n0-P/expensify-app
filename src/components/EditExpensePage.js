import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import { OptionModal } from './Modal'

export class EditExpensePage extends React.Component {

  state = {
    selectedOption: undefined
  }

  handleOpenModal = () => {
    this.setState(() => ({ selectedOption: true }));
  }

  handleCloseModal = () => {
    this.setState(() => ({ selectedOption: undefined }));
  }

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  };

  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button-blue button-blue--secondary" onClick={this.handleOpenModal}>Remove Expense</button>

          <OptionModal
          showModal={this.state.selectedOption}
          onRemove={this.onRemove}
          handleCloseModal={this.handleCloseModal}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)




