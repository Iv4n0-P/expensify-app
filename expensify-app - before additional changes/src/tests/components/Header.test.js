//react-test-renderer
//import ReactShallowRenderer from 'react-test-renderer/shallow'
import React from 'react'

//enzyme
import { shallow } from 'enzyme'
//import toJSON from 'enzyme-to-json'

//header component
import { Header } from '../../components/Header'



//react-test-renderer
/* test ('should render header correctly', () => {
    const renderer = new ReactShallowRenderer()
    renderer.render(<Header />)
    expect (renderer.getRenderOutput()).toMatchSnapshot()
}) */

//enzyme test
test ('should render Header correctly', () => {
   const wrapper = shallow(<Header startLogout={() => {}}/>)
   //expect(wrapper.find('h1').text()).toBe('Expensify')
   expect(wrapper).toMatchSnapshot()
})

test('should call startLogout on button click', () => {
    const startLogout = jest.fn()
    const wrapper = shallow(<Header startLogout={startLogout} />)
    wrapper.find('button').simulate('click')
    expect(startLogout).toHaveBeenCalled()
})