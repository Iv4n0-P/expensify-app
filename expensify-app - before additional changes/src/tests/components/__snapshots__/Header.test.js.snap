// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        activeClassName="is-active"
        className="header_title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Expensify
        </h1>
      </Link>
      <button
        className="button button-blue--nobg"
        id="btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
